tremor-script/src/grammar.lalrpop

```diff
@@ -23,7 +24,172 @@ use beef::Cow;
+    <start:@L> "define" <kind:ConnectorKind> "connector" <id:Ident> "end" <end:@L> => DeployStmtRaw::ConnectorDecl(ConnectorDeclRaw { start, end, id: id.to_string(), kind, args: vec![], params: None }),
+    // Flow - these define how pipelines and connectors are interconnected to form a unit deployment graph
+    <start:@L> "define" "flow" <id:Ident> <args:ArgSpecificationClause> <links:WithDeployLinks> "end" <end:@L> => DeployStmtRaw::FlowDecl(FlowDeclRaw { start, end, id: id.to_string(), args: args.0, params: args.1, links }),
+    <start:@L> "define" "flow" <id:Ident> <links:WithDeployLinks> "end" <end:@L> => DeployStmtRaw::FlowDecl(FlowDeclRaw { start, end, id: id.to_string(), args: vec![], params: None, links }),
+
+    <start:@L> "deploy" <id:Ident> "from" <target:ModularTarget> <args:ArgFinalizationClause> <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id, module: target.0, target: target.1, params: args }),
+    <start:@L> "deploy" <id:Ident> "from" <target:ModularTarget> <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id: id.clone(),  module: target.0, target: target.1, params: None }),
+    <start:@L> "deploy" <id:Ident> <args:ArgFinalizationClause> <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id: id.clone(),  module: vec![], target: id, params: args }),
+    <start:@L> "deploy" <id:Ident>  <end:@L> => DeployStmtRaw::CreateStmt(CreateStmtRaw { start, end, id: id.clone(), module: vec![], target: id, params: None }),
+}
+
+// Allows arguments to be specified and defaults to be provided and overridden in definitions
+ArgSpecificationClause: (WithArgsRaw<'input>, Option<DeployWithExprsRaw<'input>>) = {
+    <args:DeployArgs> <params:DeployConfigClause> => (args, Some(params)),
+    <args:DeployArgs> => (args, None),
+    <params:DeployConfigClause> => (vec![], Some(params)),
+}
+
+// Allows arguments to be consumed with defaults or overriden in deployment statements
+ArgFinalizationClause: Option<DeployWithExprsRaw<'input>> = {
+    <params:DeployWithClause> "end" => Some(params),
+}
+
+ConnectorKind: (Vec<IdentRaw<'input>>, IdentRaw<'input>) = {
+    <mt:ModularTarget> => <>,
+}
+
+DeployArgs: WithArgsRaw<'input> = {
+    "args" <with:WithArgs> => <>,
+}
+
+DeployModuleStmt: DeployModuleStmtRaw<'input> = {
+    <start:@L> "mod" <name:Ident> "with" <doc:ModComment> <stmts:DeployModuleStmts> "end" <end:@L> => DeployModuleStmtRaw{name, start, end, stmts, doc: doc},
+}
+
+WithDeployLinks: DeployLinksRaw<'input> = {
+    "links" <links:DeployLinks> => <>,
+}
+
+DeployLinks: DeployLinksRaw<'input> = {
+    <fields:DeployLinks_> => {
+        let mut fields = fields;
+        fields.reverse();
+        fields
+    },
+}
+
+DeployLinks_: DeployLinksRaw<'input> = {
+    <Sep<DeployLinks_, DeployLink, ",">> => <>
```

Comment:
this should be `";"` to be in line with the RFC and the other statements we have

Reply:
We use `,` for `args`, for `config` and for `with` and `;` to separate statements so `,` felt more consistent.
I've checked `;` and it will work here without introducing grammar ambiguity - but to me it looks odd / out of
place.

Reply:
I added a note with the alternative and we can pick one when we connectors are merged and we have the full syntax of the `connect` expressions in place with some working examples ( benchmarks, integration tests, tutorials ) ported over to the deployment language - perhaps then the better alternative will reveal itself? Defaulted to ';' per rfc but IMO the rfc is wrong and ',' is slightly better for the reasoning above

Reply:
Ja unless we need to do it otherwise we should what what was agreed on in the RFC after all that's what it's for. Why `;` seems more correct to me is that we use `;` to separate commands (in script, in troy etc) so things that "do" something, and we use d`,` to separate data (arrays, records, config, with...) `connect a to b` seems closer to `select from a to b` then it does to `a: b`

There are exceptions to this, most notably in `case` which I think we all hate :joy:  but haven't found a way to make larlpop solve this (we might want to look at that) same goes for `patch` where we sepeerate commands with `,` for some reason. (but I think changing those two is out of the scope of that)